src/Umbraco.Web/HtmlStringUtilities.cs

```diff
@@ -235,10 +247,82 @@ public IHtmlString Truncate(string html, int length, bool addElipsis, bool treat
                     outputms.Position = 0;
                     using (TextReader outputtr = new StreamReader(outputms))
                     {
-                        return new HtmlString(outputtr.ReadToEnd().Replace("  ", " ").Trim());
+                        string result = String.Empty;
+
+                        string firstTrim = outputtr.ReadToEnd().Replace("  ", " ").Trim();
+
+                        //Check to see if there is an empty char between the hellip and the output string
+                        //if there is, remove it
+                        if (String.IsNullOrEmpty(firstTrim) == false)
+                        {
+                            result = firstTrim[firstTrim.Length - hellip.Length -1] == ' ' ? firstTrim.Remove(firstTrim.Length - hellip.Length -1, 1) : firstTrim;
+                        }
+                        return new HtmlString(result);
                     }
                 }
             }
         }
+
+        /// <summary>
+        /// Returns the length of the words from a html block
+        /// </summary>
+        /// <param name="html">Html text</param>
+        /// <param name="words">Amount of words you would like to measure</param>
+        /// <returns></returns>
+        public int WordsToLength(string html, int words, bool tagsAsContent)
+        {
+            HtmlDocument doc = new HtmlDocument();
+
+            int wordCount = 0,
+                length = 0,
+                insideTagCounter = length,
+                maxWords = words;
+
+            string strippedOfTags = string.Empty;
+
+            //If tagsAsContent is on, use the string stripped of html tags
+            if (tagsAsContent == false)
+            {
+                doc.LoadHtml(html);
+
+                foreach (var node in doc.DocumentNode.ChildNodes)
+                {
+                    strippedOfTags += node.InnerText;
+                }
+                html = strippedOfTags;
+            }
+
+            while (length < html.Length)
+            {
+                // Check to see if the current wordCount reached the maxWords allowed
+                if (wordCount.Equals(maxWords)) break;
+                // Check if current char is part of a word
+                while (length < html.Length && char.IsWhiteSpace(html[length]) == false)
+                {
+                    //Check if we have a space inside a tag and increase the length if we do
+                    if (html[length].Equals('<') && html[length + 1].Equals('/') == false && tagsAsContent)
+                    {
+                        while (html[insideTagCounter].Equals('>') == false)
```

Comment:
If there is incorrectly formatted html, could this cause an infinite loop? (i.e. there won't ever be a '>')

Reply:
What a noob :/, I'll fix those above today.